spring-data-microservice-transaction-sample/README.md

```diff
@@ -0,0 +1,548 @@
+
+`executeTwoPcTransaction()` API, called on Order Service, automatically performs preparations, validations and commits of both the local Order Service and the remote Customer Serivice. These steps are executed sequentially after the above CRUD operations successfully finish. The implementations to invoke `prepare`, `validate` and `commit` gRPC endpoints of Customer Service need to be passed as parameters to the API (the code is [here](order-service/src/main/java/sample/order/OrderService.java#L134-L141)):
+
+```java
+Collections.singletonList(
+  RemotePrepareCommitPhaseOperations.createSerializable(
+    this::callPrepareEndpoint,
+    this::callValidateEndpoint,
+    this::callCommitEndpoint,
+    this::callRollbackEndpoint
+  )
+)
+```
+
+![Sequence Diagram of High Level 2PC API](images/seq-diagram-high-level-2pc-api.png)
+
+In the `prepare` endpoint of Customer Service, it resumes and prepares the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L122-L126)):
+
+```java
+execTwoPcOperation(request.getTransactionId(), false, responseObserver, "Payment", () -> {
+  // Prepare the transaction
+  customerRepository.prepare();
+});
+```
+
+The transaction is resumed in `execTwoPcOperation()` as shown above.
+
+
+In the `validate` endpoint of Customer Service, it resumes and validates the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L131-L135)):
+
+```java
+execTwoPcOperation(request.getTransactionId(), false, responseObserver, "Validate", () -> {
+  // Validate the transaction
+  customerRepository.validate();
+  return Empty.getDefaultInstance();
+});
+```
+
+In the `commit` endpoint of Customer Service, it resumes and commits the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L140-L144)):
+
+```java
+execTwoPcOperation(request.getTransactionId(), false, responseObserver, "Commit", () -> {
+  // Commit the transaction
+  customerRepository.commit();
+  return Empty.getDefaultInstance();
+});
+```
+
+### Error handling
+
+When some error happens during the transaction, the transaction will be automatically rolled back by `executeTwoPcTransaction()`. The implementation to invoke `rollback` gRPC endpoint of Customer Service also needs to be passed as a parameter to the API with other ones (the code is [here](order-service/src/main/java/sample/order/OrderService.java#L134-L141)):
+
+```java
+Collections.singletonList(
+  RemotePrepareCommitPhaseOperations.createSerializable(
+    this::callPrepareEndpoint,
+    this::callValidateEndpoint,
+    this::callCommitEndpoint,
+    this::callRollbackEndpoint
+  )
+)
+```
+
+In the `rollback` endpoint of Customer Service, it resumes and rolls back the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L149-L153)):
```

Comment:
According to a [similar sample doc](https://scalardb.scalar-labs.com/docs/latest/scalardb-samples/microservice-transaction-sample/README/#error-handling), "it" seems to refer to "Customer Service". If that is correct, can we change this sentence as follows?

```suggestion
In the `rollback` endpoint of Customer Service, the microservice resumes and rolls back the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L149-L153)):
```